cmake/dev.cmake

```diff
@@ -38,6 +38,78 @@ ELSE()
         LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wextra")
         LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-pedantic")
         LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Werror")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wdouble-promotion")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wmissing-include-dirs")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wunknown-pragmas")
+        # Higher levels (max is 5) produce some strange warnings
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wstrict-overflow=2")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wtrampolines")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wfloat-equal")
```

Comment:
Yes, those flags are only used for the tests and examples and are not provided to alpaka users.